From 708178879f7e2ba84e12e9a4b63ae86bff067bce Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Tue, 7 Aug 2018 12:14:54 -0700
Subject: [PATCH 0393/1852] xilinx: v4l: dma: Fix bug in calculating interlaced
 height

commit e1e50d1e288b65d9b84a594a323da7749dada6da from
https://github.com/Xilinx/linux-xlnx.git

In current implementation, v4l bridge driver is dividing height
by 2 in case interlaced capture. According to v4l2 specification,
application should divide height by 2 and pass to driver for
interlaced capture.

This patch will avoid the height altering calculations in driver.

Reference:
https://www.kernel.org/doc/html/v4.16/media/uapi/v4l/pixfmt-v4l2.html

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dma.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dma.c b/drivers/media/platform/xilinx/xilinx-dma.c
index c4affec3a11b..522833e81167 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-dma.c
@@ -882,8 +882,6 @@ __xvip_dma_try_format(struct xvip_dma *dma,
 				info->bpl_factor;
 		pix_mp->height = clamp(pix_mp->height, XVIP_DMA_MIN_HEIGHT,
 				       XVIP_DMA_MAX_HEIGHT);
-		if (pix_mp->field == V4L2_FIELD_ALTERNATE)
-			pix_mp->height = pix_mp->height / 2;
 
 		/*
 		 * Clamp the requested bytes per line value. If the maximum
@@ -947,9 +945,6 @@ __xvip_dma_try_format(struct xvip_dma *dma,
 		pix->height = clamp(pix->height, XVIP_DMA_MIN_HEIGHT,
 				    XVIP_DMA_MAX_HEIGHT);
 
-		if (pix->field == V4L2_FIELD_ALTERNATE)
-			pix->height = pix->height / 2;
-
 		min_bpl = (pix->width * info->bpl_factor *
 			  padding_factor_nume * bpl_nume) /
 			  (padding_factor_deno * bpl_deno);
-- 
2.31.1

